File: LeetCode/1051/1051.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        auto temp = heights;
        std::ranges::sort(temp);
        return std::ranges::count_if(
            std::views::zip(heights, temp),
            [](const auto& par){
                auto[first, second] = par;
                return first != second;
            }
        );
        
    }
};

File: LeetCode/1051/1051.rs
impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {
        let mut temp: Vec<i32> = heights.iter().cloned().collect();
        temp.sort();
        heights.iter()
        .zip(temp.iter())
        .filter(|(left, right)| left != right)
        .count() as i32
    }
}
